src/ContentEntityTranslate.php

```diff
@@ -95,6 +97,46 @@ public function translate(ContentEntityInterface $srcEntity, array $translation)
     return $targetEntity;
   }
 
+  /**
+   * Translate paragraphs and their nested children.
+   *
+   * @param \Drupal\Core\Entity\ContentEntityInterface $targetEntity
+   *   The host entity to attach the paragraph translations to.
+   * @param string $targetLangcode
+   *   The target language to save the translation for.
+   * @param array $translations
+   *   The list of paragraph translations.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   * @throws \Drupal\Core\TypedData\Exception\MissingDataException
+   * @throws \Drupal\Core\TypedData\Exception\ReadOnlyException
+   * @throws \Drupal\typed_data\Exception\InvalidArgumentException
+   */
+  public function translateParagraphs(ContentEntityInterface $targetEntity, $targetLangcode, array $translations) {
+    foreach ($targetEntity->getFields() as $fieldItemList) {
+      if ($fieldItemList instanceof EntityReferenceFieldItemListInterface && $fieldItemList->getSetting('target_type') ==='paragraph') {
+        foreach ($fieldItemList as $delta => $item) {
+          if ($item->entity) {
+            $paragraph = $item->entity;
+            foreach ($translations as $translation) {
+              $uuid = !empty($paragraph->duplicateSource) ? $paragraph->duplicateSource->uuid() : $paragraph->uuid();
+              if ($uuid === $translation['entityId']) {
+                $paragraph = $this->translate($paragraph->getTranslation($targetLangcode), $targetLangcode, $translation);
+                $this->translateParagraphs($paragraph, $targetLangcode, $translations);
+                $paragraph->setNewRevision(TRUE);
+                $paragraph->setNeedsSave(TRUE);
+                break;
+              }
+            }
+
+            $fieldItemList[$delta] = $paragraph;
```

Comment:
Q: What does overriding the `$fieldItemList[$delta]` accomplish here?

Reply:
Basically re-attaching the translated paragraph to the host node paragraphs field.

Reply:
Another way of writing could be `$item->entity = $paragraph` , but I followed the pattern of the paragraph_asymmetric_translation module

Reply:
Got it 👍